integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/RedbeamsDatabaseTest.java

```diff
@@ -0,0 +1,282 @@
+        String generatedKey = resourcePropertyProvider().getName();
+        testContext
+                .given(RedbeamsDatabaseTestDto.class)
+                .withName(databaseName)
+                .withConnectionUserName(username)
+                .withConnectionPassword(password)
+                .withConnectionURL(connectionUrl)
+                .withType(type)
+                .when(databaseTestClient.createV4(), RunningParameter.key(generatedKey))
+                .expect(BadRequestException.class,
+                        RunningParameter.expectedMessage(expectedErrorMessage)
+                                .withKey(generatedKey))
+                .validate();
+    }
+
+    /*
+    @Test(dataProvider = INVALID_ATTRIBUTE_PROVIDER)
+    public void testDatabaseTestConnectionWithInvalidAttribute(
+            TestContext testContext,
+            String databaseName,
+            String username,
+            String password,
+            String connectionUrl,
+            String expectedErrorMessage,
+            @Description TestCaseDescription testCaseDescription) {
+        String generatedKey = resourcePropertyProvider().getName();
+
+        testContext
+                .given(DatabaseTestTestDto.class)
+                .withRequest(new DatabaseV4Request())
+                .withName(databaseName)
+                .withConnectionUserName(username)
+                .withConnectionPassword(password)
+                .withConnectionURL(connectionUrl)
+                .withType("HIVE")
+                .when(databaseTestClient.testV4(), RunningParameter.key(generatedKey))
+                .expect(BadRequestException.class,
+                        RunningParameter.expectedMessage(expectedErrorMessage)
+                                .withKey(generatedKey))
+                .validate();
+    }
+    */
+
+/*
+    @DataProvider(name = DB_TYPE_PROVIDER)
+    public Object[][] provideTypes() {
+        List<DatabaseType> databaseTypeList = Arrays.asList(DatabaseType.values());
+        Object[][] objects = new Object[databaseTypeList.size()][3];
+        databaseTypeList
+                .forEach(databaseType -> {
+                    objects[databaseTypeList.indexOf(databaseType)][0] = getBean(TEST_CONTEXT_CLASS);
+                    objects[databaseTypeList.indexOf(databaseType)][1] = databaseType;
+                    objects[databaseTypeList.indexOf(databaseType)][2] =
+                            new TestCaseDescription.TestCaseDescriptionBuilder()
+                                    .given("there is a running cloudbreak")
+                                    .when("sending database create request with databaseType '" + databaseType + '\'')
+                                    .then("creation should be successful");
+                });
+        return objects;
+    }
+*/
+    @DataProvider(name = INVALID_ATTRIBUTE_PROVIDER)
+    public Object[][] provideInvalidAttributes() {
```

Comment:
This part and related test cases may can move to lower level of testing (Unit testing). If it is truthfully and reliably possible there. How about this @afarsang?

Reply:
Unit tests created, so I will remove this set of tests from the integration test layer.